_posts/2023-08-01-tmt-cross-project-testing.md

```diff
@@ -0,0 +1,131 @@
+```yaml
+# reverse dependency test
+enabled: false
+
+adjust+:
+  when: revdeps == yes
+  enabled: true
+
+discover:
+    how: fmf
+    url: https://github.com/cockpit-project/cockpit-podman
+    ref: "main"
+execute:
+    how: tmt
+
+/podman-system:
+    summary: Run cockpit-podman system tests
+    discover+:
+        test: /test/browser/system
+
+# ... possibly more test plans here
+```
+
+Conversely, you should mark the main test plan for the dependency project to *not* run in the "revdeps" scenario, to keep them apart in the PR statuses and log views:
+
+```yaml
+adjust+:
+  when: revdeps == yes
+  enabled: false
+```
+
+Finally, plug it all together: Tell `packit.yaml` to run the "revdeps" context test plan on current Fedora against your "builds from main" COPR, by specifying [tf_extra_params](https://packit.dev/docs/configuration/upstream/tests#optional-parameters). You can of course choose more/different `target`s, as long as your COPR has matching builds for them. Also give it a meaningful `identifier`, so that you can tell the statuses apart:
+
+```yaml
+  # On current Fedora, run reverse dependency tests against https://copr.fedorainfracloud.org/coprs/g/cockpit/main-builds/
+  - job: tests
+    identifier: revdeps
+    trigger: pull_request
+    targets:
+      - fedora-latest-stable
+    tf_extra_params:
+      environments:
+        - artifacts:
+          - type: repository-file
+            id: https://copr.fedorainfracloud.org/coprs/g/cockpit/main-builds/repo/fedora-$releasever/group_cockpit-main-builds-fedora-$releasever.repo
+          tmt:
+            context:
+              revdeps: "yes"
+```
+
+This will result in a new packit test status where you can see the result and logs:
+
+![packit revdeps test status](/images/packit-revdeps-test-status.png)
+
+For an example, see this [cockpit PR](https://github.com/cockpit-project/cockpit/pull/19155) which enables this. This approach was developed and tested extensively [in an experimental PR](https://github.com/cockpit-project/cockpit/pull/19117).
+
+## Conclusion
+
+This demonstrates that it is possible today to do upstream cross-project testing for reverse dependency gating, without any extra privileges, secrets, or self-managed infrastructure. I truly believe that this approach on a bigger scale will lead to fewer regressions, less frustration, and avoid long and annoying bugzilla/fix/release turnaround cycles. It can increase velocity across the FOSS landscape in the same way as pervasive reverse dependency testing increased stability and velocity in Debian and Ubuntu.
+
+Many thanks to [Karel Srot](https://github.com/kkaarreell) from tmt, [Laura Barcziová](https://github.com/lbarcziova) and [Frantisek Lachman](https://github.com/lachmanfrantisek) from Packit, and [Miroslav Vadkerti](https://github.com/thrix) from Testing Farm for their great help with figuring out all the details here!
```

Comment:
Hi Martin, this is very nice post. Great job! I also like the chosen approach where there are dedicated revdep tests and agreed terms of cooperations (as you are mentioning), instead of running everything available. Just a nitpick, I am not from `tmt` project. Even though I have contributed with few patches I definitely do not consider myself as project representative, I am more a happy user. If you will, you can mention [keylime](https://github.com/keylime/keylime) project where we are also using some of the approaches you are describing.

Reply:
Hah, thanks for spotting. Updated to "who knows a lot about tmt" :grin: